File: mcp/fdep-mcp-server/fdep_mcp/compliance_baseline.py
from __future__ import annotations

import json
from pathlib import Path
from typing import cast


SHARED_COMPLIANCE_BASELINE_PATH = (
    Path(__file__).resolve().parents[2] / "mcp-compliance" / "baseline-policy.json"
)


def read_shared_compliance_baseline() -> dict[str, object]:
    return cast(
        dict[str, object],
        json.loads(SHARED_COMPLIANCE_BASELINE_PATH.read_text(encoding="utf-8")),
    )

File: mcp/ast-grep-mcp-server/src/complianceBaseline.ts
import { readFileSync } from "node:fs";
import path from "node:path";

export const SHARED_COMPLIANCE_BASELINE_PATH = path.resolve(
  __dirname,
  "../../mcp-compliance/baseline-policy.json"
);

export function readSharedComplianceBaseline(): unknown {
  return JSON.parse(readFileSync(SHARED_COMPLIANCE_BASELINE_PATH, "utf-8"));
}
